shell/platform/darwin/ios/ios_platform_surface_gl.mm

```diff
@@ -0,0 +1,67 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/shell/platform/darwin/ios/ios_platform_surface_gl.h"
+#include "flutter/common/threads.h"
+#include "flutter/lib/ui/painting/resource_context.h"
+#include "flutter/shell/platform/darwin/ios/framework/Source/vsync_waiter_ios.h"
+#include "third_party/skia/include/core/SkSurface.h"
+#include "third_party/skia/include/gpu/GrBackendSurface.h"
+#include "third_party/skia/include/gpu/GrTexture.h"
+#include "third_party/skia/include/gpu/GrTypes.h"
+
+namespace shell {
+
+IOSPlatformSurfaceGL::~IOSPlatformSurfaceGL() {
+  // TODO(sigurdm, mravn): dispose cache
+}
+
+IOSPlatformSurfaceGL::IOSPlatformSurfaceGL(NSObject<FlutterPlatformSurface>* surface)
+    : surface_(surface) {
+  FXL_DCHECK(surface_);
+}
+
+sk_sp<SkImage> IOSPlatformSurfaceGL::MakeSkImage(int width, int height, GrContext* grContext) {
+  ASSERT_IS_GPU_THREAD;
+  CVPixelBufferRef buffer;
+  fxl::AutoResetWaitableEvent latch;
+  blink::Threads::IO()->PostTask([this, &latch, &buffer]() {
+    if (cache_ == nullptr) {
+      CVReturn err = CVOpenGLESTextureCacheCreate(kCFAllocatorDefault, NULL,
+                                                  [EAGLContext currentContext], NULL, &cache_);
+      if (err != noErr) {
+        FXL_LOG(WARNING) << "Failed to create GLES texture cache: " << err;
+      }
+    }
+    buffer = [surface_ getPixelBuffer];
```

Comment:
`fml::CFRef` for safer management of the buffer lifetime.

Reply:
Since I dont think `getPixelBuffer` has transfer-out ownership, you also need a `CFRetain` on it.

Reply:
Done.

Reply:
Update: `getPixelBuffer` is now renamed to `copyPixelBuffer` and we don't call `CFRetain`.
Done.